### PR TITLE
feat - create new post

### PR DESCRIPTION
**Pull request by prachi-sahani**
This PR includes the changes for the "create new post" functionality
- Required validation for all the input fields
- Validation for adding up to 3 tags only 
image
image
image
image

### CODE REVIEW

File: src/components/AddQuestionModal.tsx [hunk 1]
```diff
@@ -1,69 +1,205 @@
 import {
   Button,
+  LoadingButton,
   TextField,
   Autocomplete,
   Dialog,
   DialogActions,
   DialogContent,
   DialogTitle,
 } from "../utilities/material-ui/material-components";
```
Comment (agrawalshanu708) [inline src/components/AddQuestionModal.tsx:10]: great!! This is good practice to make separate classes of repetitive code.

**Review by agrawalshanu708 — APPROVED** [review]
code is well structured..
UI looks plain, add some salt to it by time...
else you can merge it

File: src/utilities/services/feed.ts [hunk 2]
```diff
@@ -57,4 +55,26 @@ const callAddVote = async (
...
 };
 
-export { callGetAllQuestions, callAddComment, callGetAllVotes, callAddVote };
+const callAddQuestion = async (
+  { token, question }: { token: string; question: NewQuestion },
+  { rejectWithValue }: any
+) => {
+  try {
+    const { data } = await addNewQuestion(token, question);
+    console.log(data);
```
Comment (junaidshaikh-js) [inline src/utilities/services/feed.ts]: remove console logs

**Review by junaidshaikh-js — APPROVED** [review]
Good to merge.

File: src/utilities/services/feed.ts (hunk 2 above)
Comment (prachi-sahani) [reply src/utilities/services/feed.ts]: Done

**Review by Hamdrive — COMMENTED** [review]
Implement for new post looks good
Clean code structure
Good to merge

**Pull request merged**